### PR TITLE
PSPDFKit Android

### PR DESCRIPTION
**Pull request by satindar**
- add pspdfkit android dep
- add large heap declaration
- wrap pspdfkit documentview in a fragment

### CODE REVIEW

**Review by jacksonh — APPROVED** [review]

File: android/Omnivore/app/src/main/java/app/omnivore/omnivore/models/Highlight.kt [hunk 1]
```diff
@@ -11,4 +20,39 @@ data class Highlight(
...
+}
+
+data class HighlightPatch(
+  val bbox: List<Float>,
+  val blendMode: String,
+  val color: String,
+  val rects: List<List<Float>>,
+  val pageIndex: Int
+) {
+  fun rectList(): List<RectF> {
```
[the remark this answers is not in the document]
Comment (satindar) [reply android/Omnivore/app/src/main/java/app/omnivore/omnivore/models/Highlight.kt:48]: Tried using the instantJSON like we do on iOS but the value comes back as null. So maybe not compatible with Android.

[the remark this answers is not in the document]
Comment (satindar) [reply android/Omnivore/app/src/main/java/app/omnivore/omnivore/models/Highlight.kt:48]: Screen Shot 2022-10-03 at 1 31 57 PM

LMAO

[the remark this answers is not in the document]
Comment (satindar) [reply android/Omnivore/app/src/main/java/app/omnivore/omnivore/models/Highlight.kt:48]: iOS for comparison

Simulator Screen Shot - iPhone 14 - 2022-10-03 at 13 32 49

[the remark this answers is not in the document]
Comment (satindar) [reply android/Omnivore/app/src/main/java/app/omnivore/omnivore/models/Highlight.kt:48]: Comparison of iOS and Android values in the rects array:

```
//    "NSRect: {{125.03699493408203, 406.92364501953125}, {225.08401489257812, 10.787994384765625}}",
//              [125.03699493408203, 194.28836059570312, 225.08401489257812, 10.787994384765625]


//    "NSRect: {{46.064994812011719, 391.92364501953125}, {170.82000732421875, 10.787994384765625}}",
//              [46.064994812011719, 209.28836059570312, 170.82000732421875, 10.787994384765625]
```

2nd value in array is off for some reason 🤔

**Comment by jacksonh** [comment]
@satindar did you figure out the instant json stuff, anything you want me to look at?

**Review by jacksonh — APPROVED** [review]

**Comment by satindar** [comment]
> @satindar did you figure out the instant json stuff, anything you want me to look at?

Yeah got the instantJSON working now so highlights are being loaded now and it matches iOS. Just have a threading issue to resolve. The pdf doesn't open 50% of the time. I think something about the way LiveData is being used to update the UI.

I think I can get that resolved today and then I'll merge this one.

**Comment by satindar** [comment]
I'm still running into a threading issue (sent a support request to PSPDFKit):

```
2022-10-06 08:56:15.332 2662-2662/app.omnivore.omnivore E/PSPDFKit.PdfView: Can't initialize fragment contents
    android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.
        at android.view.ViewRootImpl.checkThread(ViewRootImpl.java:9619)
        at android.view.ViewRootImpl.requestLayout(ViewRootImpl.java:1917)
        at android.view.View.requestLayout(View.java:26282)
        at android.view.View.requestLayout(View.java:26282)
        at android.view.View.requestLayout(View.java:26282)
        at android.view.View.requestLayout(View.java:26282)
        at android.view.View.requestLayout(View.java:26282)
        at android.view.ViewGroup.addView(ViewGroup.java:5075)
        at android.view.ViewGroup.addView(ViewGroup.java:5017)
        at android.view.ViewGroup.addView(ViewGroup.java:4989)
        at com.pspdfkit.internal.vf.<init>(SourceFile:42)
        at com.pspdfkit.internal.views.document.DocumentView.a(SourceFile:13)
        at com.pspdfkit.internal.ck.a(SourceFile:50)
        at com.pspdfkit.internal.ck.$r8$lambda$CNsJIzEjpPOo7RcNrMZjpD-uVdI(Unknown Source:0)
        at com.pspdfkit.internal.ck$$ExternalSyntheticLambda4.run(Unknown Source:6)
        at io.reactivex.internal.operators.completable.CompletableFromAction.subscribeActual(CompletableFromAction.java:35)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at io.reactivex.internal.operators.completable.CompletableSubscribeOn$SubscribeOnObserver.run(CompletableSubscribeOn.java:64)
        at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
        at com.pspdfkit.internal.fl$c.run(SourceFile:1)
        at com.pspdfkit.internal.fl.a(SourceFile:9)
        at com.pspdfkit.internal.fl.$r8$lambda$KjCEGKTggCzbgx-NO95_zQV4k_g(Unknown Source:0)
        at com.pspdfkit.internal.fl$$ExternalSyntheticLambda0.run(Unknown Source:2)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
        at java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
        at java.lang.Thread.run(Thread.java:1012)
        ```

**Pull request merged**